BasementRenovator.py

```diff
@@ -3430,60 +3426,66 @@ class EntityGroupModel(QAbstractListModel):
     def __init__(self, kind=None, entityList=None):
         QAbstractListModel.__init__(self)
 
-        self.groups = {}
+        self.groups = []
         self.kind = kind
         self.view = None
+        self.entitycount = 0
 
-        self.filter = ""
+        entitiesCounted = {}
 
-        if entityList is None:
-            entityList = xmlLookups.entities.lookup()
+        entries = entityList
 
-        for entity in entityList:
-            try:
-                imgPath = Path(entity.imagePath)
-            except:
-                traceback.print_exception(*sys.exc_info())
-                imgPath = Path()
+        global xmlLookups
+        if kind is not None:
+            entries = xmlLookups.entities.lookupGroup(kind)
+        elif entityList is None:
+            entries = xmlLookups.entities.lookup()
 
-            if not imgPath.exists():
-                printf(
-                    f"Could not find Entity {entity.name} ({entity.type}.{entity.variant}.{entity.subtype})'s palette icon:",
-                    imgPath,
-                )
-                entity.imagePath = "resources/Entities/questionmark.png"
+        i = 0
+        populatingGroup = None
 
-            entityItem = EntityItem(entity)
+        for entry in entries:
+            if isinstance(entry, xmlLookups.entities.GroupConfig):
+                if populatingGroup:
+                    populatingGroup.calculateIndices(i)
+                    i = populatingGroup.endIndex + 1
 
-            addKinds = entity.kinds
-            if self.kind is not None:
-                addKinds = [self.kind]
+                populatingGroup = EntityGroupItem(entry.label, entry.containsEntity())
```

Comment:
Yeah. I'm planning on redoing EntityGroupItem to be more of a nested structure, since that'll clean up a lot of this weird `containsEntity()` and visibility business.